main.tf

```diff
@@ -0,0 +1,82 @@
+ */
+locals {
+  addon = {
+    name = "keda-controller"
+
+    helm_chart_name    = "keda"
+    helm_chart_version = "2.14.2"
+    helm_repo_url      = "https://kedacore.github.io/charts"
+  }
+
+  addon_irsa = {
+    operator = {
+      service_account_create = var.keda_operator_service_account_create
+      service_account_name   = var.keda_operator_service_account_name
+
+      irsa_role_create         = var.keda_operator_irsa_role_create
+      irsa_policy_enabled      = var.keda_operator_irsa_policy_enabled
+      irsa_policy              = var.keda_operator_irsa_policy
+      irsa_assume_role_enabled = var.keda_operator_irsa_assume_role_enabled
+      irsa_assume_role_arns    = var.keda_operator_irsa_assume_role_arns
+      irsa_additional_policies = var.keda_operator_irsa_additional_policies
+    }
+    metricServer = {
+      service_account_create = var.keda_metric_server_service_account_create
+      service_account_name   = var.keda_metric_server_service_account_name
+
+      irsa_role_create         = var.keda_metric_server_irsa_role_create
+      irsa_policy_enabled      = var.keda_metric_server_irsa_policy_enabled
+      irsa_policy              = var.keda_metric_server_irsa_policy
+      irsa_assume_role_enabled = var.keda_metric_server_irsa_assume_role_enabled
+      irsa_assume_role_arns    = var.keda_metric_server_irsa_assume_role_arns
+      irsa_additional_policies = var.keda_metric_server_irsa_additional_policies
+    }
+    webhooks = {
+      service_account_create = var.keda_webhooks_service_account_create
+      service_account_name   = var.keda_webhooks_service_account_name
+
+      irsa_role_create         = var.keda_webhooks_irsa_role_create
+      irsa_policy_enabled      = var.keda_webhooks_irsa_policy_enabled
+      irsa_policy              = var.keda_webhooks_irsa_policy
+      irsa_assume_role_enabled = var.keda_webhooks_irsa_assume_role_enabled
+      irsa_assume_role_arns    = var.keda_webhooks_irsa_assume_role_arns
+      irsa_additional_policies = var.keda_webhooks_irsa_additional_policies
+    }
+  }
+
+  addon_values = yamlencode({
+    rbac = {
+      create = var.rbac_create != null ? var.rbac_create : true
+    }
+    serviceAccount = {
+      operator = {
+        create = var.keda_operator_service_account_create
+        name   = var.keda_operator_service_account_name
+        annotations = length(module.addon-irsa["operator"].iam_role_attributes) > 0 ? {
+          "eks.amazonaws.com/role-arn" = module.addon-irsa["operator"].iam_role_attributes.arn
+        } : tomap({})
+      }
+      metricServer = {
+        create = var.keda_metric_server_service_account_create
+        name   = var.keda_metric_server_service_account_name
+        annotations = length(module.addon-irsa["metricServer"].iam_role_attributes) > 0 ? {
+          "eks.amazonaws.com/role-arn" = module.addon-irsa["metricServer"].iam_role_attributes.arn
+        } : tomap({})
```

Comment:
Nice-to-have:

I'm wondering if a condition based on the output of the module in the form of length won't cause problems. I understand that this condition involves variables `var.enabled && var.rbac_create && var.service_account_create && var.irsa_role_create` which is uniq per configuration section (except var.enabled and var.rbac_create). The question is whether there is an alternative way to evaluate the condition so that we don't rely on the module's output.

Reply:
Reworked this using a new output variable irsa_role_enabled from the addon-irsa module